ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,353 @@
+                string message = succes && (input < sizeLimit) ? "Id get successfully input" : "Invalid input Try again";
+                Console.WriteLine(message);
+            }
+            while (!succes);
+
+            return input;
+        }
+
+        static void DeleteSwap(string[] records, int id)
+        {
+            if (id == records.Length)
+                return;
+            string swap = records[^1];
+            records[^1] = records[id];
+            records[id] = swap;
+        }
+        public enum OrderBy
+        {
+            Ascending,
+            Descending
+        }
+
+        public enum SortBy
+        {
+            Name,
+            Surname,
+            Phone
+        }
+        static void Sort(string path)
+        {
+            Console.WriteLine("Enter sorting order: ");
+            Console.WriteLine("1. Ascending");
+            Console.WriteLine("2. Descending");
+            ConsoleKey pressed = Console.ReadKey().Key;
+            OrderBy order = (pressed == ConsoleKey.D2) ? OrderBy.Descending : OrderBy.Ascending;
+
+            Console.WriteLine("Sorting by: ");
+            Console.WriteLine("1. Name");
+            Console.WriteLine("2. Surname");
+            Console.WriteLine("3. Phone");
+            pressed = Console.ReadKey().Key;
+
+            SortBy sort;
+            if (pressed == ConsoleKey.D3)
+            {
+                sort = SortBy.Phone;
+            }
+            else if (pressed == ConsoleKey.D2)
+            {
+                sort = SortBy.Surname;
+            }
+            else
+            {
+                sort = SortBy.Name;
+            }
+            var records = ReadPhoneBookRecords(path);
+
+            BubbleSort(records, order, sort);
+            string[] recordsToWrite = new string[records.Length + 1];
+            recordsToWrite[0] = "FirstName,LastName,PhoneNumber";
+            for (int i = 1; i < recordsToWrite.Length; i++)
+            {
+                recordsToWrite[i] = records[i - 1].Item1 + "," + records[i - 1].Item2 + "," + records[i - 1].Item3;
+            }
+            File.WriteAllLines(path, recordsToWrite);
+        }
+        static void SortBinary(string path)
+        {
+            var records = ReadPhoneBookRecords(path);
+
+            BubbleSort(records, OrderBy.Ascending, SortBy.Surname);
+            string[] recordsToWrite = new string[records.Length + 1];
+            recordsToWrite[0] = "FirstName,LastName,PhoneNumber";
+            for (int i = 1; i < recordsToWrite.Length; i++)
+            {
+                recordsToWrite[i] = records[i - 1].Item1 + "," + records[i - 1].Item2 + "," + records[i - 1].Item3;
+            }
+            File.WriteAllLines(path, recordsToWrite);
+        }
+
+        public static void BubbleSort((string name, string surname, string phone)[] array, OrderBy order, SortBy criteria)
+        {
+            switch (criteria)
```

Comment:
You have almost identical 3 parts, but the only sort field decision  is different

Reply:
changed to a smaller version.